sql/table_cache.cc

```diff
@@ -123,16 +123,15 @@ struct Table_cache_instance
     records, Share_free_tables::List (TABLE::prev and TABLE::next),
     TABLE::in_use.
   */
-  mysql_mutex_t LOCK_table_cache;
+  /** Avoid false sharing between instances */
+  mysql_mutex_t LOCK_table_cache MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE);
```

Comment:
As the comment says: false sharing between instances has to be avoided. False sharing within instance is more or less fine. Thus a proper target for ```MY_ALIGNED()``` is ```struct Table_cache_instance``` itself, not ```LOCK_table_cache```.

Even if the above is fixed, ```Table_cache_instance``` is allocated with ```new []```, which does not return a pointer aligned on cache line size.

When the above is fixed, please also remove 1 extra table cache instance allocated just to avoid false sharing (see ```tdc_init()```).